Realm/RLMQueryUtil.mm

```diff
@@ -85,6 +85,34 @@ BOOL RLMPropertyTypeIsNumeric(RLMPropertyType propertyType) {
     }
 }
 
+
+// Declare an overload set using lambdas or other function objects.
+// A minimal version of C++ Library Evolution Working Group proposal P0051R2.
+
+template <typename Fn, typename... Fns>
+struct Overloaded : Fn, Overloaded<Fns...> {
+    template <typename U, typename... Rest>
+    Overloaded(U&& fn, Rest&&... rest) : Fn(std::forward<U>(fn)), Overloaded<Fns...>(std::forward<Rest>(rest)...) { }
+
+    using Fn::operator();
+    using Overloaded<Fns...>::operator();
+};
+
+template <typename Fn>
+struct Overloaded<Fn> : Fn {
+    template <typename U>
+    Overloaded(U&& fn) : Fn(std::forward<U>(fn)) { }
+
+    using Fn::operator();
+};
+
+template <typename... Fns>
+Overloaded<Fns...> overload(Fns&&... f)
```

Comment:
There's probably a better home for this. Not sure where, though. Object store's `util` directory, perhaps?